src/ui/interaction/files/AvFileUpload/AvFileUpload.vue

```diff
@@ -187,33 +235,58 @@ function onDragLeave () {
 }
 
 function onChange ($event: InputEvent) {
-  emit('update:modelValue', ($event.target as HTMLInputElement)?.value)
-  emit('change', ($event.target as (InputEvent['target'] & { files: FileList }))?.files)
+  const files = ($event.target as HTMLInputElement).files
+  emit('change', files as FileList)
+  emit('update:modelValue', files?.[0] ?? null)
+}
+
+const isPreview = computed(() => props.fileName || (props.modelValue && !props.enableMultiple))
+
+const uploadLabelAttrs = computed(() => {
+  return {
+    'for':
+    props.id,
+    'class':
+    [
+      'fr-upload-group',
+      {
+        'fr-upload-group--error': props.error,
+        'fr-upload-group--valid': props.validMessage,
+        'fr-upload-group--disabled': props.disabled,
+        'drag-over': isDragging.value,
+      },
+    ],
+    'aria-label':
+    props.ariaLabel,
+    'onDragover':
+    onDragOver,
+    'onDragleave':
+    onDragLeave,
+    'onDrop':
+    onDrop,
+  }
+})
+
+function onClear (modelValue: File | null) {
+  if (modelValue) {
+    emit('update:modelValue', null)
+    emit('update:validMessage', null)
+    emit('update:error', null)
+    emit('change', [] as unknown as FileList)
+  }
+  else {
+    props.onDeleteFile?.()
+  }
 }
 </script>
 
 <template>
-  <label
-    :for="id"
-    class="fr-upload-group"
-    :class="{
-      'fr-upload-group--error': error,
-      'fr-upload-group--valid': validMessage,
-      'fr-upload-group--disabled': disabled,
-      'drag-over': isDragging,
-    }"
-    :aria-label="ariaLabel"
-    @dragover="onDragOver"
-    @dragleave="onDragLeave"
-    @drop="onDrop"
+  <component
+    :is="isPreview ? 'div' : 'label'"
+    v-bind="isPreview ? {} : uploadLabelAttrs"
+    :class="isPreview ? 'file-preview-container' : ''"
   >
-    <div
-      class="file-upload-container"
-      :class="{
-        'file-upload-container--error': error,
-        'file-upload-container--valid': validMessage,
-      }"
-    >
+    <div :class="!isPreview ? 'file-upload-container' : ''">
```

Comment:
```suggestion
    <div :class="isPreview ? '' : 'file-upload-container'">
```
Pour garder une certaine logique